docs/source/inference.rst

```diff
@@ -0,0 +1,463 @@
++-------------------+----------------+------------------------------+
+| Benchmark         | Phi-4 mini-Ins | Phi-4-mini-instruct-float8dq |
++===================+================+==============================+
+| Peak Memory (GB)  | 8.91           | 5.70 (36% reduction)         |
++-------------------+----------------+------------------------------+
+
+
+**Latency Benchmarking**:
+
+.. code-block:: bash
+
+    # baseline
+    python benchmarks/benchmark_latency.py --input-len 256 --output-len 256 --model microsoft/Phi-4-mini-instruct --batch-size 1
+
+    # float8dq
+    VLLM_DISABLE_COMPILE_CACHE=1 python benchmarks/benchmark_latency.py --input-len 256 --output-len 256 --model pytorch/Phi-4-mini-instruct-float8dq --batch-size 1
+
+**Serving Benchmarking**:
+
+We benchmarked the throughput in a serving environment.
+
+.. code-block:: bash
+
+    # Download sharegpt dataset:
+    wget https://huggingface.co/datasets/anon8231489123/ShareGPT_Vicuna_unfiltered/resolve/main/ShareGPT_V3_unfiltered_cleaned_split.json
+
+    # Other datasets can be found in: https://github.com/vllm-project/vllm/tree/main/benchmarks
+    # Note: you can change the number of prompts to be benchmarked with --num-prompts argument for benchmark_serving script.
+
+    # For baseline
+    # Server:
+    vllm serve microsoft/Phi-4-mini-instruct --tokenizer microsoft/Phi-4-mini-instruct -O3
+    # Client:
+    python benchmarks/benchmark_serving.py --backend vllm --dataset-name sharegpt --tokenizer microsoft/Phi-4-mini-instruct --dataset-path ./ShareGPT_V3_unfiltered_cleaned_split.json --model microsoft/Phi-4-mini-instruct --num-prompts 1
+
+    # For float8dq
+    # Server:
+    VLLM_DISABLE_COMPILE_CACHE=1 vllm serve pytorch/Phi-4-mini-instruct-float8dq --tokenizer microsoft/Phi-4-mini-instruct -O3
+    # Client:
+    python benchmarks/benchmark_serving.py --backend vllm --dataset-name sharegpt --tokenizer microsoft/Phi-4-mini-instruct --dataset-path ./ShareGPT_V3_unfiltered_cleaned_split.json --model pytorch/Phi-4-mini-instruct-float8dq --num-prompts 1
+
+
+**Results (H100 machine)**
+
++----------------------------+----------------+------------------------------+
+| Benchmark                  | Phi-4 mini-Ins | Phi-4-mini-instruct-float8dq |
++============================+================+==============================+
+| latency (batch_size=1)     | 1.64s          | 1.41s (1.16x speedup)        |
++----------------------------+----------------+------------------------------+
+| latency (batch_size=128)   | 3.1s           | 2.72s (1.14x speedup)        |
++----------------------------+----------------+------------------------------+
+| serving (num_prompts=1)    | 1.35 req/s     | 1.57 req/s (1.16x speedup)   |
++----------------------------+----------------+------------------------------+
+| serving (num_prompts=1000) | 66.68 req/s    | 80.53 req/s (1.21x speedup)  |
++----------------------------+----------------+------------------------------+
+
+
+Sparsity Integration
```

Comment:
this should not be a separate section I think, it can be merged into `Float8 Dynamic Quantization` section, and just mention for more quantization/sparsity, please see https://huggingface.co/docs/transformers/main/en/quantization/torchao